pkgs/development/libraries/opencl-headers/fix-cmakelists-darwin.patch

```diff
@@ -0,0 +1,25 @@
+--- a/CMakeLists.txt	2022-07-25 23:02:49.388511074 -0400
++++ b/CMakeLists.txt	2022-07-25 23:03:21.785394845 -0400
+@@ -22,8 +22,8 @@
+ 
+ if(CMAKE_SYSTEM_NAME MATCHES Darwin)
+   execute_process(
+-    COMMAND ln -shf ${CMAKE_CURRENT_SOURCE_DIR}/CL ${CMAKE_CURRENT_BINARY_DIR}/CL
+-    COMMAND ln -shf ${CMAKE_CURRENT_SOURCE_DIR}/CL ${CMAKE_CURRENT_BINARY_DIR}/OpenCL
++    COMMAND ln -snf ${CMAKE_CURRENT_SOURCE_DIR}/CL ${CMAKE_CURRENT_BINARY_DIR}/CL
```

Comment:
Could you also send this patch upstream as well?
I have not seen a `-h` flag for ln so far.

Reply:
Ok. Seems to come from macos. But macos also supports `-n`.

Reply:
Yes, `-h` it's a BSD `ln` flag, that's why I had to patch it. It supports `-n` for compatibility, so it works in both BSD and GNU versions.

And also yes, I could send them upstream, but laziness was stronger, I was doing a few more PR in here.

Reply:
I may send the patch in the next few days though, I do have it in mind.

Reply:
see https://github.com/KhronosGroup/OpenCL-Headers/pull/203